File: 2023/js/6.js
const s = Date.now();
const total = find(204171312101780, 34908986);
const e = Date.now();

console.log(total);
console.log(e - s);

function find(distance, time)
{
	const start = Math.floor(time / 2);
	var current = 0;
    var total = 1;
    var diff = 1;
    while (total > current) {
        current = total;
        [diff, -diff].forEach(offset => {
        	if ((start + offset) * (time - (start + offset)) > distance) {
                total++;
            }
        });
        diff++;
    }
    return total;
}

File: 2023/python/6.py
#!/usr/bin/python3

import math
import time

def find(distance, time):
    start = math.floor(time / 2)
    current = 0
    total = 1
    diff = 1
    while (total > current):
        current = total
        for offset in [diff, -diff]:
            if ((start + offset) * (time - (start + offset)) > distance):
                total += 1
        diff += 1
    return total

s = time.time()
total = find(204171312101780, 34908986)
e = time.time()
print(total)
print((e - s) * 1000);
